### PR TITLE
pios_servo: dma-dshot-- improve factoring

### PR DESCRIPTION
**Pull request by mlyle**
Untested.  Does this look plausible?

### CODE REVIEW

File: flight/PiOS/STM32/pios_servo.c [hunk 1]
```diff
@@ -738,7 +743,7 @@ static int DSHOT_Update()
 
 	switch (dshot_mode) {
 		case UNCONFIGURED:
-			return -1;
+			return !triggered;
```
Comment (tracernz) [inline flight/PiOS/STM32/pios_servo.c:746]: I am triggered by this.

**Review by tracernz — COMMENTED** [review]
Seems reasonable.

File: flight/PiOS/STM32/pios_servo.c (hunk 1 above)
Comment (mlyle) [reply flight/PiOS/STM32/pios_servo.c:746]: :<

**Comment by mlyle** [comment]
@glowtape can you please look at this and see if you think it's correct?

**Comment by glowtape** [comment]
Looks good to me. Tested it on the bench on the Seppuku with my scope, both in stock configuration and with pins 1-2 gimped to GPIO, works as it should.

**Comment by mlyle** [comment]
Thx

**Comment by mlyle** [comment]
Should be able to merge-- this is a small part of #1953 that people have been flying a lot.

**Review by tracernz — APPROVED** [review]

**Comment by mlyle** [comment]
Thx

**Pull request merged**